### DISCUSSION TITLE
Use of Wiki Updater Tasks in Azure DevOps

### DISCUSSION THREAD

**Discussion by sorrentinov** [rfennell/AzurePipelines#1292] [General]
Hi,

When I create a software release I would to use the Wiki Updater Task update my release history wiki page (after generate  my markdown file in build pipeline using your Generate Release plugin in)
The problem is related to "Add task". I'm unable to add Wiki Updater Task event though if it is installed: 

![Screenshot 2022-04-26 at 12 53 18](https://user-images.githubusercontent.com/4456300/165285278-0423def8-707e-4fce-b7ed-6bccc6993d9d.png)

Can you suggest me what I wrong?

Regards
Vincenzo

**Comment by rfennell** [comment]
The WIKI Updater Tasks extension contains two tasks, one for single files and another for folders

So, you need to pick the `Git based WIKI single file updater from Black Marble ' as shown at the top of the page.

Check [this video](https://www.youtube.com/embed/xaV3dFoQdV8) on the way I use the tasks

**Reply by sorrentinov** [reply]
Thanks very much!

I've just a question related to the linked work Items associated with a Build and how to see them in a Release
I have a sprint (Eg. Sprint 1) with an epic and different feature work items associated (Eg. WI1, WI2,  WI3)

If I do a Build pipeline I see - in the markdown generated - a build number XXX with WI1, WI2, WI3 item associated
When I do a Release pipeline (after build), even though I select check stage to "true" I do not found the work Items WI1, WI2, WI23.

How can I preserve the work items associated in Release? I thought that the checkStage=true permitted to compare the old stage (the first time no exist WI) with new stage

**Reply by rfennell** [reply]
'CheckStage` parameter makes a task looks for the last successful release of the pipeline to the stage containing the release notes tasks and attempts to make its version comparisons from there. 

It sounds like you are using Classic Build and Release (the UI based version). So, I suspect the problem you have is if you have a  build1 and you release it (release1) you would expect to see all the WI associated with Build1, but if you release it again i.e. Release2 that re-release Buld1 you will get no associated WIs.This is because Build1 has not changed.

It is easy to get into strange states like this with Classic Build and Release - I find life much easier with multistage YAML pipelines as they do both the build and release.

The best tip is - If you run the pipeline with diagnostics on (the check box when you queue the build or a variable set to `system.debug=true`) the task's log shows all the analysis as to how it finds associations.This usually explains what you are seeing

**Reply by sorrentinov** [reply]
Thanks for your suggestions

Yes I'm using Classic Build, I've to move on YAML as you said and will use diagnostic.
However my problem is little different.

1. If use a build pipeline WITH Generate and Update wiki task (to update Build Notes wiki page) and a release pipeline (starting this  artifact) containing Generate and Update wiki task (to update Release Notes wiki page) I see the work items on Build Notes but not on Release Notes
2. If I use build pipeline WITHOUT Generate and Update wiki task and a release pipeline starting from the build pipeline as artifact containing Generate and Update wiki task (to update Release Notes wiki page) I see the work items on Release Notes  

Moreover It seems that I'm only getting work items from the latest build associated with my release. I feel like I should be getting work items for all builds since the last release (in my Release Notes wiki page). Is there any way to get this list?

**Reply by rfennell** [reply]
> Moreover It seems that I'm only getting work items from the latest build associated with my release. 

Yes - in a release you only get WI associated with the latest build - it is a limitation

This is why I like to use your Option 1 and get the builds to update a WIKI (and/or create a [PDF](https://marketplace.visualstudio.com/items?itemName=richardfennellBM.BM-VSTS-WikiPDFExport-Tasks) of the changes. I thnk ship this to interested parties.
